feat(AI Chat): Implement authoring view

geoffreykwan opened the pull request:
## Changes
- Implemented authoring view for the AI Chat item

Note: I will create another PR for restricting who can create an AI Chat item

## Test
- AI Chat "Add New Component", "More Info" demo
- Adding an AI Chat item
- Authoring the parameters of the AI Chat item
- Previewing the AI Chat item using "Preview Component"
- Changing the model in the AI Chat authoring advanced view

geoffreykwan committed:
feat(AI Chat): Implement authoring view

hirokiterashima commented on src/assets/wise5/components/aiChat/ai-chat-authoring/ai-chat-authoring.component.ts:
```diff
@@ -0,0 +1,9 @@
+import { AbstractComponentAuthoring } from '../../../authoringTool/components/AbstractComponentAuthoring';
+
+@Component({
+  selector: 'ai-chat-authoring',
```
Not used, remove?
```suggestion
```

geoffreykwan replied:
We don't remove the selector for any of the other item authoring components. Should we be consistent?

hirokiterashima commented on src/assets/wise5/components/aiChat/edit-ai-chat-advanced/edit-ai-chat-advanced.component.ts:
```diff
@@ -0,0 +1,12 @@
+@Component({
+  selector: 'edit-ai-chat-advanced',
+  templateUrl: './edit-ai-chat-advanced.component.html',
+  styleUrls: ['./edit-ai-chat-advanced.component.scss']
```
Scss is an empty file, remove this line and the file?
```suggestion
```

hirokiterashima reviewed (commented):
Works well.

Should we make the model into a select dropdown in the advanced view?

It'd be good to work on the phrasing a bit in the hints for the System Prompt and help them thing about writing a good system prompt...maybe add some examples and tips for what makes a good system prompt? We should have hints for the prompt too, to distinguish between it and the System Prompt and the computer avatar fields.

Also, I added some code improvement suggestion comments inline.

geoffreykwan committed:
feat(AI Chat): Add drop down to select model

geoffreykwan committed:
feat(AI Chat): Remove unused files

hirokiterashima commented:
New changes look good. 

Should EditAiChatAdvancedComponent.models be a protected field?

Let's work on the the hints for the inputs.

geoffreykwan committed:
chore(AI Chat): Add protected modifier to variable

geoffreykwan committed:
feat(AI Chat): Add system prompt help dialog

hirokiterashima commented:
Good to have more info about the system prompt. I'm wondering if it should be a "show/hide more info below" kind of UX instead of the popup? It might be good to be able to see the explanation and examples as the author is typing. Let's discuss.

breity commented:
@hirokiterashima That's a good point. I agree that show/hide for more info might be more useful here. Yes, let's discuss.

I'm also wondering how we should handle authoring input fields that have hints once multiple language units are supported. We will need to make sure both hints (the informational ones like we have in this PR and the one showing the default language text) are displayed when user is translating into a different language. But we'll probably also want to distinguish them visually somehow.

geoffreykwan committed:
feat(AI Chat): Toggle system prompt help below system prompt textarea

geoffreykwan commented:
I've moved the system prompt instructions and example to below the system prompt textarea.

geoffreykwan committed:
feat(AI Chat): Move system prompt instructions out of mat hint

Move system prompt instructions above the system prompt.
Add prompt instructions above the prompt.

geoffreykwan commented:
I've moved the system prompt instructions out of the mat-hint and placed it above the system prompt. I also added prompt instructions above the prompt.

hirokiterashima reviewed (approved):
LGTM

breity committed:
Edit instructions text and make minor style updates

breity reviewed (approved):
LGTM. 👍 

I made some minor updates to the instructions text and styles.
